### PR TITLE
Skills endpoint Error Handling

### PR DESCRIPTION
**Pull request by mahigangal**
## 📝 Description

Updated /skills to return a clear 404 when no insights are recorded (missing or empty insights file), narrowed exception handling, and added tests for the new behaviours and edge cases. Reviews noted that an empty 200 is confusing when the insights file doesn’t exist or has no entries. The change clarifies this state and adds tests to lock in the behaviour.

**Closes:** #331 

---

## 🔧 Type of Change

- [x] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [ ] ♻️ Refactoring
- [ ] ⚡ Performance improvement

---

## 🧪 Testing

python -m pytest test/test_skills_API.py

Added more tests to handle edge cases and more robust error handling.

---

## ✓ Checklist

- [x] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [ ] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [x] 🔗 Any dependent changes have been merged and published in downstream modules
- [x] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots

> 

<details>
<summary>Click to expand screenshots</summary>
Screenshot 2026-01-30 at 8 39 31 PM

</details>

### CODE REVIEW

File: test/test_skills_API.py
Comment (ThunderIW) [inline test/test_skills_API.py]: Image
TEST pass with no issue

File: test/test_skills_API.py [hunk 1]
```diff
@@ -54,12 +61,110 @@ def test_list_skills_detailed(self) -> None:
...
+            with patch.object(skills_API.runtimeAppContext, "legacy_save_dir", Path(tmpdir)):
+                with patch("src.API.skills_API.list_skill_history") as mock_history:
+                    mock_history.return_value = []
+                    response = self.client.get("/skills")
+
+        self.assertEqual(response.status_code, 404)
+        self.assertEqual(
+            response.json().get("detail"),
+            "No project insights have been recorded yet.",
+        )
```
Comment (ThunderIW) [inline test/test_skills_API.py]: Good test — validates the core behavior change from this PR. Verifies the 404 fires when the file simply doesn't exist on disk.

File: test/test_skills_API.py [hunk 2]
```diff
@@ -54,12 +61,110 @@ def test_list_skills_detailed(self) -> None:
...
+        with tempfile.TemporaryDirectory() as tmpdir:
+            storage_path = Path(tmpdir) / "project_insights.json"
+            storage_path.write_text("[]", encoding="utf-8")
+            with patch.object(skills_API.runtimeAppContext, "legacy_save_dir", Path(tmpdir)):
+                with patch("src.API.skills_API.list_skill_history") as mock_history:
+                    mock_history.side_effect = OSError("boom")
+                    response = client.get("/skills")
+
+        self.assertEqual(response.status_code, 500)
+        self.assertIn("Failed to retrieve skills:", response.json().get("detail", ""))
```
Comment (ThunderIW) [inline test/test_skills_API.py]: These Pairs of test are good, proving that the 500 path still works for the specific exception types they are catching

File: src/API/skills_API.py [hunk 3]
```diff
@@ -20,14 +20,26 @@ def list_skills(detailed: bool = False) -> list:
...
         history = list_skill_history(storage_path=storage_path)
-    except Exception as exc:
+        if not history:
+            raise HTTPException(
+                status_code=404,
+                detail="No project insights have been recorded yet.",
+            )
+    except HTTPException:
+        raise
+    except (TypeError, OSError) as exc:
```
Comment (ThunderIW) [inline src/API/skills_API.py]: And make the expect catch a specific types is good  so that it only raise error on these errors, this is nicely done

**Review by ThunderIW — APPROVED** [review]
LGTM. Everything looks well made and the tests pass with no issues.
